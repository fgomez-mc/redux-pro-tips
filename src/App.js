import {albumsMock} from './albums.mock';
import Album from './components/Album/Album';

function App() {
  return (
    <div className="App">
      <h2>Albums</h2>
      <div>
        {albumsMock.map(({ id, title }) => (
          <Album key={id} title={title} id={id} />
        ))}
      </div>
    </div>
  );
}

export default App;
