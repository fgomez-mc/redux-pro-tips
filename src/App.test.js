import renderer from "react-test-renderer";
import * as redux from "react-redux";
import App from "./App";

describe("App", () => {
  jest.spyOn(redux, "useDispatch");
  const useSelectorMock = jest.spyOn(redux, "useSelector");
  const albums = Array(6)
    .fill(null)
    .map((_, i) => ({ id: i, title: `Album #${i}` }));

  test("renders learn react link", async () => {
    useSelectorMock.mockReturnValueOnce(albums);
    const testRenderer = await renderer.create(<App />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
