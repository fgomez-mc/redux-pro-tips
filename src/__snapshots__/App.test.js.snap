// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders learn react link 1`] = `
<div
  className="App"
>
  <h2>
    Albums
  </h2>
  <div>
    <h5
      className="Album"
    >
      0
      ) 
      Album #0
    </h5>
    <h5
      className="Album"
    >
      1
      ) 
      Album #1
    </h5>
    <h5
      className="Album"
    >
      2
      ) 
      Album #2
    </h5>
    <h5
      className="Album"
    >
      3
      ) 
      Album #3
    </h5>
    <h5
      className="Album"
    >
      4
      ) 
      Album #4
    </h5>
    <h5
      className="Album"
    >
      5
      ) 
      Album #5
    </h5>
  </div>
</div>
`;
